src/LinuxPerf.jl

```diff
@@ -350,18 +350,24 @@ end
 const PR_TASK_PERF_EVENTS_DISABLE = Cint(31)
 const PR_TASK_PERF_EVENTS_ENABLE = Cint(32)
 
-# syscall is lower overhead than calling libc's prctl
-function enable_all!()
-    SYS_prctl == -1 && error("prctl error : unknown architecture")
-    res = ccall(:syscall, Cint, (Clong, Clong...), SYS_prctl, PR_TASK_PERF_EVENTS_ENABLE)
-    Base.systemerror(:prctl, res < 0)
-end
-function disable_all!()
-    SYS_prctl == -1 && error("prctl error : unknown architecture")
-    res = ccall(:syscall, Cint, (Clong, Clong...), SYS_prctl, PR_TASK_PERF_EVENTS_DISABLE)
+@inline function fast_prctl(op)
+    res = if SYS_prctl == -1
+        ccall(:prctl, Cint, (Cint, Cint...), op)
+    elseif Sys.ARCH == :x86_64
+        Base.llvmcall("""
+        %a = call i32 asm sideeffect "syscall", "={rax},{rax},{rdi},~{rcx},~{r11},~{memory}"(i64 %0, i32 %1)
+        ret i32 %a
+        """, Int32, Tuple{Int64, Int32}, SYS_prctl, op)
```

Comment:
```suggestion
    elseif Sys.ARCH == :x86_64
        Base.llvmcall("""
        %a = call i64 asm sideeffect "syscall", "={rax},{rax},{rdi},~{rcx},~{r11},~{memory}"(i64 %0, i64 %1)
        ret i64 %a
        """, Int64, Tuple{Int64, Int64}, SYS_prctl, Int64(op))
```

I don't think this affects the code at all, but the register size mismatches concern me